03-attribute-operations.qmd

```diff
@@ -92,17 +91,29 @@ In both cases the method is followed by square brackets, and two indices, separa
 * A `list`, as in `[0,2,4]`
 * `:`—indicating "all" indices
 
-The once exception which we are going to with subsetting by indices is when selecting columns, directly using a list, as in `df[['a','b']]`, instead of `df.loc[:, ['a','b']]`, to select columns `'a'` and `'b'` from `df`.
+An exception to this rule is selecting columns using a list, as in `df[['a','b']]`, instead of `df.loc[:, ['a','b']]`, to select columns `'a'` and `'b'` from `df`.
 
 Here are few examples of subsetting the `GeoDataFrame` of world countries.
 
-Subsetting rows by position:
+Subsetting rows by position, e.g., the first three rows:
 
 ```{python}
 world.iloc[0:3, :]
 ```
 
-Subsetting columns by position:
+which is equivalent to:
+
+```{python}
+world.iloc[:3]
+```
+
+as well as:
+
+```{python}
+world.head(3)
+```
+
+Subsetting columns by position, e.g., the first three columns:
```

Comment:
I'm not sure we need to mention more methods than the bare minimum (that is, unless they are essential and used in the book), to not confuse the reader

Reply:
I think iloc[0:3, :] is the "scariest" of the three options. For me, it is much harder to remember than head(3) or just iloc[:3] and I would never use it if I don't absolutely have to.

Reply:
Interesting, thanks! I can understand your point of view. Again, this is perhaps subjective, but I prefer to keep in mind one explicit and general method, even if sometimes it makes the code longer or less elegant than using a shortcut. At least when teaching the material, to make things simpler, but in daily work often as well. That is, I prefer to use `dat[i, j]` because it's a one-fits-all solution (only assuming that I remember that `i` is for rows and `j` is for columns), rather than keeping in mind *that* plus several shortcuts, such as `.head` or the fact that `.iloc[:3]` refers to rows, etc. I think that @Nowosad 's suggestion is great in this context: we can keep things simple (using minimal set of methods) in the main text, while adding information boxes giving the various shortcuts and alternatives.